### PR TITLE
nginx: update to 1.25.3

### PR DESCRIPTION
**Pull request by tiagogaspar8**
Maintainer: Thomas Heil <heil@terminal-consulting.de> Christian Marangi <ansuelsmth@gmail.com>
Compile tested: X86 Fedora
Run tested: Ax3600

Description:
Update nginx to the latest version.

### CODE REVIEW

**Comment by jefferyto** [comment]
The powerpc/riscv64 build errors are because of luajit2 - ping @Ansuel, @BKPepe

powerpc_464fp log:

```
DYNLINK   libluajit.so
/builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/../../../../powerpc-openwrt-linux-musl/bin/ld: bss-plt forced due to lj_vm.o
/builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/../../../../powerpc-openwrt-linux-musl/bin/ld: warning: libluajit.so has a LOAD segment with RWX permissions
/builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/../../../../powerpc-openwrt-linux-musl/bin/ld: /builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/crtbeginS.o: in function `__do_global_dtors_aux':
crtstuff.c:(.text+0x46): R_PPC_PLT16_HA relocation unsupported for bss-plt
/builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/../../../../powerpc-openwrt-linux-musl/bin/ld: crtstuff.c:(.text+0x4e): R_PPC_PLT16_LO relocation unsupported for bss-plt
/builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/../../../../powerpc-openwrt-linux-musl/bin/ld: crtstuff.c:(.text+0x50): R_PPC_PLTSEQ relocation unsupported for bss-plt
/builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/../../../../powerpc-openwrt-linux-musl/bin/ld: crtstuff.c:(.text+0x54): R_PPC_PLTCALL relocation unsupported for bss-plt
/builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/../../../../powerpc-openwrt-linux-musl/bin/ld: crtstuff.c:(.text+0x6a): R_PPC_PLT16_HA relocation unsupported for bss-plt
/builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/../../../../powerpc-openwrt-linux-musl/bin/ld: crtstuff.c:(.text+0x6e): R_PPC_PLT16_LO relocation unsupported for bss-plt
/builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/../../../../powerpc-openwrt-linux-musl/bin/ld: crtstuff.c:(.text+0x70): R_PPC_PLTSEQ relocation unsupported for bss-plt
/builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/../../../../powerpc-openwrt-linux-musl/bin/ld: crtstuff.c:(.text+0x74): R_PPC_PLTCALL relocation unsupported for bss-plt
/builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/../../../../powerpc-openwrt-linux-musl/bin/ld: /builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/crtbeginS.o: in function `frame_dummy':
crtstuff.c:(.text+0xca): R_PPC_PLT16_HA relocation unsupported for bss-plt
/builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/../../../../powerpc-openwrt-linux-musl/bin/ld: crtstuff.c:(.text+0xd2): R_PPC_PLT16_LO relocation unsupported for bss-plt
/builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/../../../../powerpc-openwrt-linux-musl/bin/ld: crtstuff.c:(.text+0xd4): R_PPC_PLTSEQ relocation unsupported for bss-plt
/builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/../../../../powerpc-openwrt-linux-musl/bin/ld: crtstuff.c:(.text+0xd8): R_PPC_PLTCALL relocation unsupported for bss-plt
/builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/../../../../powerpc-openwrt-linux-musl/bin/ld: lj_err.o: in function `err_unwind':
luajit2-2.1-20231006/src/lj_err.c:200:(.text+0x2ba): R_PPC_PLT16_HA relocation unsupported for bss-plt
/builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/../../../../powerpc-openwrt-linux-musl/bin/ld: luajit2-2.1-20231006/src/lj_err.c:200:(.text+0x2be): R_PPC_PLT16_LO relocation unsupported for bss-plt
/builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/../../../../powerpc-openwrt-linux-musl/bin/ld: luajit2-2.1-20231006/src/lj_err.c:200:(.text+0x2c0): R_PPC_PLTSEQ relocation unsupported for bss-plt
/builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/../../../../powerpc-openwrt-linux-musl/bin/ld: luajit2-2.1-20231006/src/lj_err.c:200:(.text+0x2c4): R_PPC_PLTCALL relocation unsupported for bss-plt
[snip thousands of lines]
/builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/../../../../powerpc-openwrt-linux-musl/bin/ld: lib_init.o: in function `luaL_openlibs':
luajit2-2.1-20231006/src/lib_init.c:43:(.text+0x2a): R_PPC_PLT16_HA relocation unsupported for bss-plt
/builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/../../../../powerpc-openwrt-linux-musl/bin/ld: luajit2-2.1-20231006/src/lib_init.c:44:(.text+0x2e): R_PPC_PLT16_HA relocation unsupported for bss-plt
/builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/../../../../powerpc-openwrt-linux-musl/bin/ld: luajit2-2.1-20231006/src/lib_init.c:45:(.text+0x32): R_PPC_PLT16_HA relocation unsupported for bss-plt
/builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/../../../../powerpc-openwrt-linux-musl/bin/ld: luajit2-2.1-20231006/src/lib_init.c:47:(.text+0x56): R_PPC_PLT16_HA relocation unsupported for bss-plt
/builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/../../../../powerpc-openwrt-linux-musl/bin/ld: luajit2-2.1-20231006/src/lib_init.c:47:(.text+0x5a): R_PPC_PLT16_LO relocation unsupported for bss-plt
/builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/../../../../powerpc-openwrt-linux-musl/bin/ld: luajit2-2.1-20231006/src/lib_init.c:47:(.text+0x5c): R_PPC_PLTSEQ relocation unsupported for bss-plt
/builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/../../../../powerpc-openwrt-linux-musl/bin/ld: luajit2-2.1-20231006/src/lib_init.c:50:(.text+0x66): R_PPC_PLT16_HA relocation unsupported for bss-plt
/builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/../../../../powerpc-openwrt-linux-musl/bin/ld: luajit2-2.1-20231006/src/lib_init.c:51:(.text+0x6a): R_PPC_PLT16_HA relocation unsupported for bss-plt
/builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/../../../../powerpc-openwrt-linux-musl/bin/ld: luajit2-2.1-20231006/src/lib_init.c:47:(.text+0x6c): R_PPC_PLTCALL relocation unsupported for bss-plt
/builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/../../../../powerpc-openwrt-linux-musl/bin/ld: luajit2-2.1-20231006/src/lib_init.c:53:(.text+0x86): R_PPC_PLT16_HA relocation unsupported for bss-plt
/builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/../../../../powerpc-openwrt-linux-musl/bin/ld: luajit2-2.1-20231006/src/lib_init.c:53:(.text+0x8a): R_PPC_PLT16_LO relocation unsupported for bss-plt
/builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/../../../../powerpc-openwrt-linux-musl/bin/ld: luajit2-2.1-20231006/src/lib_init.c:53:(.text+0x8c): R_PPC_PLTSEQ relocation unsupported for bss-plt
/builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/../../../../powerpc-openwrt-linux-musl/bin/ld: luajit2-2.1-20231006/src/lib_init.c:53:(.text+0x90): R_PPC_PLTCALL relocation unsupported for bss-plt
/builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/../../../../powerpc-openwrt-linux-musl/bin/ld: luajit2-2.1-20231006/src/lib_init.c:43:(.text+0xca): R_PPC_PLT16_LO relocation unsupported for bss-plt
/builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/../../../../powerpc-openwrt-linux-musl/bin/ld: luajit2-2.1-20231006/src/lib_init.c:43:(.text+0xcc): R_PPC_PLTSEQ relocation unsupported for bss-plt
/builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/../../../../powerpc-openwrt-linux-musl/bin/ld: luajit2-2.1-20231006/src/lib_init.c:43:(.text+0xd0): R_PPC_PLTCALL relocation unsupported for bss-plt
/builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/../../../../powerpc-openwrt-linux-musl/bin/ld: luajit2-2.1-20231006/src/lib_init.c:44:(.text+0xde): R_PPC_PLT16_LO relocation unsupported for bss-plt
/builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/../../../../powerpc-openwrt-linux-musl/bin/ld: luajit2-2.1-20231006/src/lib_init.c:44:(.text+0xe0): R_PPC_PLTSEQ relocation unsupported for bss-plt
/builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/../../../../powerpc-openwrt-linux-musl/bin/ld: luajit2-2.1-20231006/src/lib_init.c:44:(.text+0xe4): R_PPC_PLTCALL relocation unsupported for bss-plt
/builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/../../../../powerpc-openwrt-linux-musl/bin/ld: luajit2-2.1-20231006/src/lib_init.c:45:(.text+0xf6): R_PPC_PLT16_LO relocation unsupported for bss-plt
/builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/../../../../powerpc-openwrt-linux-musl/bin/ld: luajit2-2.1-20231006/src/lib_init.c:45:(.text+0xf8): R_PPC_PLTSEQ relocation unsupported for bss-plt
/builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/../../../../powerpc-openwrt-linux-musl/bin/ld: luajit2-2.1-20231006/src/lib_init.c:45:(.text+0x100): R_PPC_PLTCALL relocation unsupported for bss-plt
/builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/../../../../powerpc-openwrt-linux-musl/bin/ld: luajit2-2.1-20231006/src/lib_init.c:50:(.text+0x112): R_PPC_PLT16_LO relocation unsupported for bss-plt
/builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/../../../../powerpc-openwrt-linux-musl/bin/ld: luajit2-2.1-20231006/src/lib_init.c:50:(.text+0x114): R_PPC_PLTSEQ relocation unsupported for bss-plt
/builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/../../../../powerpc-openwrt-linux-musl/bin/ld: luajit2-2.1-20231006/src/lib_init.c:50:(.text+0x118): R_PPC_PLTCALL relocation unsupported for bss-plt
/builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/../../../../powerpc-openwrt-linux-musl/bin/ld: luajit2-2.1-20231006/src/lib_init.c:51:(.text+0x12a): R_PPC_PLT16_LO relocation unsupported for bss-plt
/builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/../../../../powerpc-openwrt-linux-musl/bin/ld: luajit2-2.1-20231006/src/lib_init.c:51:(.text+0x12c): R_PPC_PLTSEQ relocation unsupported for bss-plt
/builder/staging_dir/toolchain-powerpc_464fp_gcc-12.3.0_musl/bin/../lib/gcc/powerpc-openwrt-linux-musl/12.3.0/../../../../powerpc-openwrt-linux-musl/bin/ld: luajit2-2.1-20231006/src/lib_init.c:51:(.text+0x134): R_PPC_PLTCALL relocation unsupported for bss-plt
collect2: error: ld returned 1 exit status
make[4]: *** [Makefile:753: libluajit.so] Error 1
make[4]: Leaving directory '/builder/build_dir/target-powerpc_464fp_musl/luajit2-2.1-20231006/src'
make[3]: *** [Makefile:125: default] Error 2
make[3]: Leaving directory '/builder/build_dir/target-powerpc_464fp_musl/luajit2-2.1-20231006'
make[2]: *** [Makefile:98: /builder/build_dir/target-powerpc_464fp_musl/luajit2-2.1-20231006/.built] Error 2
make[2]: Leaving directory '/feed/lang/luajit2'
time: package/feeds/packages_ci/luajit2/compile#25.35#2.50#28.91
```

riscv64 log:
```
make[3]: Entering directory '/builder/build_dir/target-riscv64_riscv64_musl/luajit2-2.1-20231006'
==== Building LuaJIT 2.1 ====
make -C src
make[4]: Entering directory '/builder/build_dir/target-riscv64_riscv64_musl/luajit2-2.1-20231006/src'
lj_arch.h:73:2: error: #error "Architecture not supported (in this version), see: https://luajit.org/status.html#architectures"
   73 | #error "Architecture not supported (in this version), see: https://luajit.org/status.html#architectures"
      |  ^~~~~
lj_arch.h:475:2: error: #error "No target architecture defined"
  475 | #error "No target architecture defined"
      |  ^~~~~
Makefile:273: *** Unsupported target architecture.  Stop.
make[4]: Leaving directory '/builder/build_dir/target-riscv64_riscv64_musl/luajit2-2.1-20231006/src'
make[3]: *** [Makefile:125: default] Error 2
make[3]: Leaving directory '/builder/build_dir/target-riscv64_riscv64_musl/luajit2-2.1-20231006'
make[2]: *** [Makefile:98: /builder/build_dir/target-riscv64_riscv64_musl/luajit2-2.1-20231006/.built] Error 2
make[2]: Leaving directory '/feed/lang/luajit2'
time: package/feeds/packages_ci/luajit2/compile#0.40#0.34#0.78
```

If the luajit package isn't installed (with `./scripts/feeds install luajit`) then `HAS_LUAJIT_ARCH` won't be defined.

**Comment by tiagogaspar8** [comment]
Rebased, can we get this merged?

File: net/nginx/Makefile [hunk 1]
```diff
@@ -507,4 +507,4 @@ define Package/nginx-mod-luci-ssl/install
 	$(INSTALL_DIR) $(1)/usr/bin
 endef
 
-$(eval $(call BuildPackage,nginx-mod-luci-ssl))
+$(eval $(call BuildPackage,nginx-mod-luci-ssl))
```
Comment (BKPepe) [inline net/nginx/Makefile:510]: ??

Comment (tiagogaspar8) [reply net/nginx/Makefile:510]: Good question, I'll see about it in an hour.

**Comment by BKPepe** [comment]
Lets merge it and solve luajit2 later.

**Pull request merged**